source/attribute/AttributeAccess.h

```diff
@@ -0,0 +1,100 @@
+/* AttributeAccess.h
+Copyright (c) 2023 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef ATTRIBUTE_ACCESS_H_
+#define ATTRIBUTE_ACCESS_H_
+
+#include "AttributeCategory.h"
+#include "AttributeEffectType.h"
+
+#include <optional>
+
```

Comment:
This class could also use a comment here to tell what it is used for; Identifies a Categorized attribute that we want to access.